src/basho_bench_worker.erl

```diff
@@ -334,3 +349,27 @@ rate_worker_run_loop(State, Lambda) ->
         ExitReason ->
             exit(ExitReason)
     end.
+
```

Comment:
What do you think of the idea of moving all of the generator logic into a self contained function so we minimize the internals of the key/val gen logic and simplify changing them up later? Something like:

```
add_generators(State, Id) ->
    KeyGen = init_generators(ApiPassState, basho_bench_config:get(key_generator), Id, basho_bench_keygen),
    ValGen = init_generators(ApiPassState, basho_bench_config:get(value_generator), Id, basho_bench_valgen),
     State#state{key_gen=KeyGen, val_gen=ValGen}.
```

and then up above in init you would just do `State1 = add_generators(State0, Id),`.